number/src/traits.rs

```diff
@@ -1,6 +1,7 @@
 use std::{fmt, hash::Hash, ops::*, str::FromStr};
 
 use num_traits::{One, Zero};
+use serde::{de::DeserializeOwned, Serialize};
```

Comment:
Is there a problem that the other structs use `Deserialize` and this one uses `DeserializeOwned`?

Reply:
To use `Deserialize` on this trait would introduce a lifetime `'de` that would propagate on every use of the `FieldElement<'de>`, because here we add a new bound to the trait, but we don't actually implement `Deserialize` for it

Reply:
Yea I've had to use this before too, just not sure if it conflicts/works well with the others

Reply:
It shouldn't, it just indicates that every implementors of `FieldElement` should have owned data (If I recall correctly)